pallets/frequency-tx-payment/src/payment.rs

```diff
@@ -35,11 +41,6 @@ where
 		key: &T::AccountId,
 		fee: Self::Balance,
 	) -> Result<Self::Balance, TransactionValidityError> {
-		ensure!(
```

Comment:
@enddynayn I'm not sure why this is removed. Any idea why?
 Edit: It looks like it is moved to the `can_withdraw_fee` but I don't see any problem just keeping it here as is

Reply:
I suppose it's just because it's redundant/inefficient (since it's in the `validate` path, and hence could be called many more times than the `pre_dispatch` patch?

Reply:
I believe this is likely due to the can_withdraw_x function being called before the fee is withdrawn.